packages/pharos/src/components/table/pharos-table.ts

```diff
@@ -0,0 +1,208 @@
+import { PharosElement } from '../base/pharos-element';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import { html, nothing } from 'lit';
+import { state } from 'lit/decorators.js';
+import type { TemplateResult, CSSResultArray } from 'lit';
+import { property } from 'lit/decorators.js';
+import { tableStyles } from './pharos-table.css';
+import { PharosSelect } from '../select/pharos-select';
+import { PharosPagination } from '../pagination/pharos-pagination';
+
+export type ColumnSpecification = {
+  name: string;
+  field: string;
+};
+
+export type RowData = {
+  [key: string]: unknown;
+};
+
+/**
+ * Pharos table component.
+ *
+ * @tag pharos-table
+ *
+ * @fires pharos-table-prev-page - Fires when the table goes to previous page
+ * @fires pharos-table-next-page - Fires when the table goes to next page
+ *
+ */
+export class PharosTable extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-select': PharosSelect,
+    'pharos-pagination': PharosPagination,
+  };
+
+  /**
+   * Column specification.
+   * @attr columns
+   * sample column:
+   * [
+   *  {
+   *    name: "ID"
+   *    field: "id"
+   *  },
+   *  {
+   *    name: "Message"
+   *    field: "message"
+   *  }
+   * ]
+   */
+  @property({ type: Array, reflect: true, attribute: 'columns' })
+  public columns: ColumnSpecification[] = [];
+
+  /**
+   * Row data.
+   * @attr row-data
+   * Example:
+   * [
+   *  {
+   *    id: 1,
+   *    message: "first one"
+   *  },
+   *  {
+   *    id: 2,
+   *    message: "second one"
+   *  }
+   * ]
+   */
+  @property({ type: Array, reflect: true, attribute: 'row-data' })
+  public rowData: RowData[] = [];
+
+  @property({ type: Boolean, reflect: true, attribute: 'hide-pagination' })
+  public hidePagination: boolean = true;
+
+  @property({ type: Number, reflect: true, attribute: 'total-results' })
+  public totalResults: number = 0;
+
+  @property({ type: Array, reflect: true, attribute: 'page-size-options' })
+  public pageSizeOptions: number[] = [50, 100];
+
+  @property({ type: String, reflect: true, attribute: 'caption' })
+  public caption: string = '';
```

Comment:
I understand we need the caption for accessibility reasons, I'm curious if there should be any ability to visually hide it in certain cases or is the expectation should be having the practice of always showing a table caption?

Reply:
☝️ @sirrah-tam might have some insight into this, I think it could be "visually hidden" as long as it is still accessible to screen readers

Reply:
It would be great to be able to include an attribute that auto-visually-hides the caption as that does appear to be pretty common in other examples I'm finding. Similar to how the <title> element gives a SVG it's accessible name (accNam), the <caption> gives the Table it's accName, helpful when users of AT are reviewing the contents of a page. Visually hiding the captions would mean that a sighted user would have to use other information on the page to understand the context and purpose of the table, ex page title or heading preceding the table.

Reply:
If I understand correctly, we want to add another attribute, something like `auto-visually-hide = true`, to control if we want to visually hide the caption? Does that mean we should make the caption attribute a required attribute so that we always have a accessible name for the table?

Reply:
I do feel like the caption should be required to enforce always having accessible tables generated but would leave that up to @sirrah-tam and the other wikkies to officially answer yes/no.

Reply:
Updated!

Reply:
The caption element is not required per WCAG and there are some cases where the caption may not be needed, especially for simple tables where there is already a heading that precedes the table. However, I do like that we'd kind of take a stand that all tables need a caption by requiring it for our design system. I would however say we shouldn't have it hidden by default and instead make the user explicitly hide it.